### PR TITLE
book 4: Fix conflated animal form and list. Add consistency with earl…

### PR DESCRIPTION
**Pull request by JoeShep**
…ier chapters.

A number of additions and changes from recent updates introduced broken code examples and confusion over which component the focus was supposed to be on in the affected chapters ( FORMS_CONTROLLED_COMPONENT and MULTIPLE_PROVIDERS ). I believe I have straightened out the form vs list issues.

I've introduced a 'thought experiment' at the end of MULTIPLE_PROVIDERS to address the fact that the example code doesn't take into consideration the fact that the animal provider uses `_expand=location`. I'm hoping that will lead to some lively discussion and a better understanding of the ways to approach joining data, ie in the query vs in the code.

### CODE REVIEW

File: book-4-nashville-kennels/chapters/FORMS_CONTROLLED_COMPONENT.md [hunk 1]
```diff
@@ -28,48 +28,59 @@ In React, we add event listeners directly on a button's `onClick` attribute.
...
-				return <AnimalCard key={animal.id} animal={animal} />
-			})
-        }
+          <h2>Animals</h2>
+		      <button onClick={() => {history.push("/animals/create")}}>
+            Add Animal
+          </button>
+          {
+            animals.map(animal => {
+              const owner = customers.find(c => c.id === animal.customerId)
```
Comment (askingalot) [inline book-4-nashville-kennels/chapters/FORMS_CONTROLLED_COMPONENT.md:52]: Do we need to specify where `customers` comes from?

File: book-4-nashville-kennels/chapters/FORMS_CONTROLLED_COMPONENT.md [hunk 2]
```diff
@@ -135,8 +153,8 @@ export const AnimalForm = () => {
       const locationId = parseInt(animal.locationId)
       const customerId = parseInt(animal.customerId)
```
Comment (askingalot) [inline book-4-nashville-kennels/chapters/FORMS_CONTROLLED_COMPONENT.md:154]: I don't think we need these `parseInt()` calls any more

**Review by askingalot — CHANGES_REQUESTED** [review]
These changes really iron out the wrinkles of these chapters. I only have a couple of comments.

File: book-4-nashville-kennels/chapters/FORMS_CONTROLLED_COMPONENT.md (hunk 2 above)
Comment (JoeShep) [reply book-4-nashville-kennels/chapters/FORMS_CONTROLLED_COMPONENT.md:154]: Good call.

File: book-4-nashville-kennels/chapters/FORMS_CONTROLLED_COMPONENT.md (hunk 1 above)
Comment (JoeShep) [reply book-4-nashville-kennels/chapters/FORMS_CONTROLLED_COMPONENT.md:52]: It's laid out in the previous chapter, but couldn't hurt to show it again here. I'll expand the example code to make it clear where we're defining customers

File: book-4-nashville-kennels/chapters/MULTIPLE_PROVIDERS.md [hunk 3]
```diff
@@ -94,7 +92,11 @@ The last step is to extract the new `customer` and `animal` keys on the object p
...
-Then display the name property of each one.
-
-Now the **`AnimalForm`** component can access data from all of the data providers to produce dropdown options.
+Then display the name property of each one.  
 ![animal card showing name of customer and location](./images/animals-after-join.png)
+
+### _Expand_ your thinking
+You may have noticed something weird in this chapter about the `locations` state. Mainly, it isn't really needed. Why? Think back to when you created the **AnimalProvider** component. In `getAnimals` you use `_expand=location`. Do you remember what that does? Why does it make it unnecessary to use the **LocationContext** in this component? 
+
+Which approach seems 'better' to you? Why? Could we also remove the need for the **OwnerContext**? How? If you want to talk this out with an instructor, grab one and hash it out! Or, group up with some classmates and think outloud with them. We would love to hear what you come up with.
```
Comment (askingalot) [inline book-4-nashville-kennels/chapters/MULTIPLE_PROVIDERS.md:101]: I missed this earlier, but you should use "CustoermContext" instead of "OwnerContext"

**Review by askingalot — CHANGES_REQUESTED** [review]

**Pull request merged**